R/gs_power_ahr.R

```diff
@@ -172,21 +172,27 @@ gs_power_ahr <- function(
   info_scale <- match.arg(info_scale)
 
   # Check if it is two-sided design or not
-  if (identical(lower, gs_b) && (!is.list(lpar))) {
+  if ((identical(lower, gs_b) && (!is.list(lpar))) || all(!test_lower)) {
     if (all(test_lower == FALSE)) {
       two_sided <- FALSE
-      lpar <- rep(-Inf, n_analysis)
     } else {
       two_sided <- ifelse(identical(lpar, rep(-Inf, n_analysis)), FALSE, TRUE)
     }
   } else {
     two_sided <- TRUE
   }
 
+  if (!two_sided) {
+    lpar <- rep(-Inf, n_analysis)
+    lower <- gs_b
+  }
+
   # Check if user input the total spending for futility,
   # if they use spending function for futility
-  if (identical(lower, gs_spending_bound) && is.null(lpar$total_spend)) {
-    stop("gs_power_ahr: please input the total_spend to the futility spending function.")
+  if (two_sided && identical(lower, gs_spending_bound)) {
+    if (is.null(lpar$total_spend) && any(test_lower)) {
```

Comment:
There are many `if`s and `&&`s in this PR. It would be nice to add some developer tests to ensure this is working as expected (and that it keeps working). If you can provide me some example code, I could convert these to developer tests (in a follow-up PR).

Reply:
Sure, I will provide some tests examples with you in a follow-up PR.